### PR TITLE
Feature/setup route

### PR DESCRIPTION
**Pull request by bdebever**
In this branch, inspired from the colleagues from last year, I have added the ability to restrict the requests to authentified users. 

Few things that will need to be done later: 
- Add the link to the work of the team in charge of the user management/login management
- Turn this on in our codebase `line 53: backend/src/api/auth.js` , `line 22: backend/src/index.js`

### CODE REVIEW

File: backend/src/api/auth.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+/**
+ * Express middleware to handle Authorization header, check token, and populate req.user with
+ * the user informations. It will use userFactory if provided with the token data, and wait
+ * for the result. The result or the token data will be populated into req.user field.
+ * If authentication fails, it won't block requests, but req.user will be undefined.
+ * If you want to protect route, use **requireAuth** middleware.
+ *
+ * @param {String} provider Linkapp URL
+ * @param {Function} [userFactory] method called with token user data
```
Comment (NansD) [inline backend/src/api/auth.js:17]: why are there brackets here ?

File: frontend/src/services/auth.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        ...headers,
+        ...{
+            Authorization: `JWT ${token}`
+        }
+    }
+);
+
+const query = { ...qs.parse(document.location.search.slice(1)) };
+if (query.token) {
+    console.log('New token received from query : ' + query.token);
```
Comment (NansD) [inline frontend/src/services/auth.js]: remove this please

File: frontend/src/services/auth.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    if ((response.status || response.response.status) === 401) {
+        const body = typeof (response.json) === 'function' ? (await response.json()) : (
+            response.data ||
+            response.body ||
+            (response.response && (response.response.data || response.response.body))
+        );
+        window.location.replace(
+            `${body.login}?redirect=${encodeURIComponent(document.location.href)}`
+        );
+        await never();
```
Comment (NansD) [inline frontend/src/services/auth.js:56]: Did you get further info about never() ? I remember we talked about it

File: frontend/src/services/login.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
+export const isLoggedIn = (props) => {
+    const { location } = props;
+    const user = localStorage.getItem('user');
+    if (!user) {
+        login();
+    }
+    console.log(user);
+    console.log(location);
```
Comment (NansD) [inline frontend/src/services/login.js]: remove this please

File: backend/src/api/auth.js (hunk 1 above)
Comment (bdebever) [reply backend/src/api/auth.js:17]: Optional parameter: https://github.com/swagger-api/swagger-codegen/issues/6345

File: frontend/src/services/auth.js (hunk 3 above)
Comment (bdebever) [reply frontend/src/services/auth.js:56]: it's resolved immediately but actually kind of useless AFAIK cauz we leave the page anyway

**Review by FelixDelcourt — APPROVED** [review]

**Comment by NansD** [comment]
Since I merged @ValentinChion's branch, there is one merge conflict. Can you look after it @bdebever ?

**Comment by NansD** [comment]
Actually I took care of it ;)

**Pull request merged**